### ISSUE TITLE
Negative enum Values

### ISSUE THREAD

**Issue by cubed-it** [rpgmaker/NetJSON#223]
Hey there,
unfortunately I found a new problem with the current version. Negative enum values cannot be deserialized when using useenumstrings = false.

Here is a simple sample:
```
   class Program
   {
      static void Main(string[] args)
      {
         NetJSON.NetJSONSettings.CurrentSettings.UseEnumString = false;
         var content = NetJSON.NetJSON.Serialize(new TestA() { EnumAProp = EnumA.ValA });
         var obj = NetJSON.NetJSON.Deserialize<TestA>(content);
      }
   }

   public enum EnumA
   {
      ValA= -1
   }

   public class TestA
   {
      public EnumA EnumAProp { get; set; }
   }
```

```
System.ArgumentException: "Must specify valid information for parsing in the string."
   at System.Enum.EnumResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument)
   at System.Enum.TryParseEnum(Type enumType, String value, Boolean ignoreCase, EnumResult& parseResult)
   at System.Enum.Parse(Type enumType, String value, Boolean ignoreCase)
   at NetJSON.Internals.SerializerUtilities.FlagStringToEnum[T](String value)
   at SetTestA(Char* , Int32& , TestA , String , NetJSONSettings )
   at CreateClassOrDictTestA(Char* , Int32& , NetJSONSettings )
   at ReadTestA(String , NetJSONSettings )
   at DeserializeValue(String )
   at NetJsonTest.Program.Main(String[] args)
```

**Comment by rpgmaker** [comment]
Thanks I will look into it.

**Comment by cubed-it** [comment]
Can you roughly estimate when you will find the time?
Would be helpful for my planning. Thanks!

**Comment by rpgmaker** [comment]
It is resolved now. Should be available in nuget.org shortly. I just published it as 1.3.1

https://www.nuget.org/packages/NetJSON/

**Comment by cubed-it** [comment]
Great, thank you very much!!
